File: leetcode-python-go/775. 全局倒置与局部倒置/main.go
package main

import (
	"fmt"
)

func isIdealPermutation(nums []int) bool {
	n := len(nums)
	minSuf := nums[n-1]
	for i := n - 2; i > 0; i-- {
		if nums[i-1] > minSuf {
			return false
		}
		minSuf = min(minSuf, nums[i])
	}
	return true
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func main() {
	nums := []int{1, 0, 2}
	results := isIdealPermutation(nums)
	fmt.Println(results)
}

File: leetcode-python-go/775. 全局倒置与局部倒置/main.py
from typing import List

nums = [1,0,2]

def isIdealPermutation(nums: List[int]) -> bool:
    min_suf = nums[-1]
    for i in range(len(nums) - 2, 0, -1):
        if nums[i - 1] > min_suf:
            return False
        min_suf = min(min_suf, nums[i])
    return True

print(isIdealPermutation(nums))